org.eclipse.winery.frontends/app/topologymodeler/src/app/canvas/canvas.component.ts

```diff
@@ -1580,6 +1582,7 @@ export class CanvasComponent implements OnInit, OnDestroy, OnChanges, AfterViewI
             }
         });
         this.entityTypes.relationshipTypes.forEach(value => this.assignRelTypes(value));
+        this.policyService.addNewPolicyEvent.subscribe(data => this.toggleModalHandler(data));
```

Comment:
This should be handled by the redux store, if a new policy is added, a corresponding action must be triggered which changes the store. A corresponding subscription detects the changes and changes the UI accordingly.